software/core/Infrastructure/backend/externalSocketParser/parser.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * The function converts the String receiving from Odroid or JSON from Angular 
+ * and return JSON Object 
+ * @param {String} externalJson: String object recieving from the Odroid or JSON 
+ * object from Angular
+ * @returns {Object}; return JSON Object for manipulation by backend
+ */
+
+const JSONParser = (externalJson) => {
+    if (
+        typeof externalJson === "string" &&
+        externalJson !== null 
+    )
+
+    {
+        let dataPreparation = externalJson.replace(/'/g, '"');
+        let JSONobject = JSON.parse(dataPreparation);
+        return JSONobject;
+
+    } else {
```

Comment:
Is there a reason for an else clause here? Can a json be a type other than string?